pallets/route-executor/src/lib.rs

```diff
@@ -575,6 +612,45 @@ impl<T: Config> Pallet<T> {
 		Ok(route)
 	}
 
+	pub fn disable_ed_handling_for_insufficient_assets(
+		skip_ed_disabling: &mut bool,
+		route_length: usize,
+		trade_index: usize,
+		trade: Trade<T::AssetId>,
+	) {
+		if route_length > 1
+			&& (!T::InspectRegistry::is_sufficient(trade.asset_in)
+				|| !T::InspectRegistry::is_sufficient(trade.asset_out))
+		{
+			*skip_ed_disabling = true;
+			//We optimize to set the state for middle trades only once at the first middle trade, then we change no state till the last trade
+			match trade_index {
+				0 => SkipEd::<T>::put(SkipEdState::SkipEdLock),
+				trade_index if trade_index.saturating_add(1) == route_length => {
+					SkipEd::<T>::put(SkipEdState::SkipEdUnlock)
+				}
+				1 => SkipEd::<T>::put(SkipEdState::SkipEdLockAndUnlock),
```

Comment:
shouldn't this be a default, not just 1 for all the cases where you are in the middle of the route?

Reply:
We would have to set another state for routes with only 1 trade if this would be default storage value.
This state is necessary only when you are in the middle of the route